readthedocs/projects/views/private.py

```diff
@@ -347,7 +348,57 @@ def get_form_kwargs(self, step=None):
 
     def get_template_names(self):
         """Return template names based on step name."""
-        return "projects/import_{}.html".format(self.steps.current)
+        return f"projects/import_{self.steps.current}.html"
+
+    def process_step(self, form):
+        # pylint: disable=too-many-nested-blocks
+        if isinstance(form, ProjectBasicsForm):
+            remote_repository = form.cleaned_data.get("remote_repository")
+            if remote_repository and remote_repository.vcs_provider == GITHUB:
+                remote_repository_relations = (
+                    remote_repository.remote_repository_relation.filter(
+                        user=self.request.user,
+                        account__isnull=False,
+                    )
+                    .select_related("account", "user")
+                    .only("user", "account")
+                )
+                for relation in remote_repository_relations:
+                    service = GitHubService(relation.user, relation.account)
+                    session = service.get_session()
+
+                    for yaml in [
+                        ".readthedocs.yaml",
+                        ".readthedocs.yml",
+                        "readthedocs.yaml",
+                        "readthedocs.yml",
+                    ]:
+                        try:
+                            response = session.head(
+                                f"https://api.github.com/repos/{remote_repository.full_name}/contents/{yaml}",
+                                timeout=1,
+                            )
+                            if response.ok:
+                                log.info(
+                                    "Read the Docs YAML file found for this repository.",
+                                    yaml=yaml,
+                                )
+                                messages.success(
+                                    self.request,
+                                    _(
+                                        "We detected a configuration file in your repository; triggering an initial build"
+                                    ),
+                                )
+                                self.form_list.pop("config")
+                                break
+                        except Exception:
+                            log.warning(
+                                "Failed when hitting GitHub API to check for .readthedocs.yaml file.",
+                                filename=yaml,
+                            )
+                            continue
```

Comment:
With a 1 second timeout, we don't need to worry about this blocking the request at least, but I could see this being a better interaction at the remote repository selection, and maybe something we could even work into an async request through that API.